server/src/compression.rs

```diff
@@ -468,11 +545,48 @@ mod tests {
 
     #[test]
     fn test_try_compress_regular_linear_time_series() {
-        let timestamps = Vec::from_iter((100..1000).step_by(100));
-        let values = Vec::from_iter((10..100).step_by(10).map(|v| v as f32));
+        let values = comp_test_util::generate_values(
+            TRY_COMPRESS_TEST_LENGTH,
+            comp_test_util::DataType::Linear,
+            None,
+            None,
+        );
+        let timestamps = comp_test_util::generate_timestamps(values.len(), false);
+        let (uncompressed_timestamps, uncompressed_values) =
+            create_uncompressed_time_series(&timestamps, &values);
+        let error_bound = ErrorBound::try_new(ERROR_BOUND_ZERO).unwrap();
+
+        let compressed_record_batch = try_compress(
+            1,
+            &uncompressed_timestamps,
+            &uncompressed_values,
+            error_bound,
+            &test_util::compressed_schema(),
+        )
+        .unwrap();
+
+        assert_compressed_record_batch_with_segments_from_regular_time_series(
+            &uncompressed_timestamps,
+            &compressed_record_batch,
+            &[models::SWING_ID],
+        )
+    }
+
+    #[test]
+    fn test_try_compress_irregular_linear_time_series() {
+        // Create a random linear equation and generate random timestamps.
+        let a: i64 = thread_rng().gen_range(-10..10);
+        let b: i64 = thread_rng().gen_range(1..50);
+        let timestamps = comp_test_util::generate_timestamps(TRY_COMPRESS_TEST_LENGTH, true);
+
+        // Calculate the corresponding values on the y axis. These values are not f32 initially
```

Comment:
Unclear why this is necessary when both `test_try_compress_regular_linear_time_series()` and `test_try_compress_regular_almost_linear_time_series()` uses `generate_values()`.

Reply:
Fixed.